File: app-mvp/src/main/java/xuk/imaginary/gui/nets/news/NewsListPresenter.java
package xuk.imaginary.gui.nets.news;

import io.reactivex.android.schedulers.AndroidSchedulers;
import xuk.imaginary.data.source.NetsRepository;
import xuk.imaginary.util.Utils;

/**
 * @author xuie
 * @date 17-7-5
 */
public class NewsListPresenter implements NewsListContract.Presenter {
    private NetsRepository netsRepository;
    private NewsListContract.View netsView;
    private int currentPage = 0;

    public NewsListPresenter(NetsRepository netsRepository, NewsListContract.View netsView) {
        this.netsRepository = Utils.checkNotNull(netsRepository);
        this.netsView = Utils.checkNotNull(netsView);
        this.netsView.setPresenter(this);
    }

    @Override
    public void subscribe() {
        getList(true);
    }

    @Override
    public void unsubscribe() {

    }

    @Override
    public void getList(boolean isRefresh) {
        if (isRefresh) {
            currentPage = 0;
        } else {
            currentPage += 20;
        }
        netsRepository.getNews(currentPage)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(netsSummaries -> netsView.addList(isRefresh, netsSummaries)
                        , Throwable::printStackTrace);

    }
}

File: app-mvp-kotlin/src/main/java/xuk/imaginary/gui/nets/news/NewsListPresenter.kt
package xuk.imaginary.gui.nets.news

import io.reactivex.android.schedulers.AndroidSchedulers
import xuk.imaginary.data.source.NETSRepository

/**
 * Created by xuie on 17-7-5.
 */

class NewsListPresenter(
    private val netsRepository: NETSRepository,
    private val netsView: NewsListContract.View
) : NewsListContract.Presenter {
  private var currentPage = 0

  init {
    this.netsView.presenter = this
  }

  override fun start() {
    getList(true)
  }

  override fun getList(isRefresh: Boolean) {
    if (isRefresh) {
      currentPage = 0
    } else {
      currentPage += 20
    }
    netsRepository.getNews(currentPage)
        .observeOn(AndroidSchedulers.mainThread())
        .subscribe({ netsSummaries ->
          netsView.addList(isRefresh, netsSummaries)
        }, { e -> e.printStackTrace() })

  }
}
